packages/ui/src/components/atoms/ExternalLink.tsx

```diff
@@ -3,47 +3,97 @@ import React from 'react';
 import { Bullet } from './Bullet';
 import { Container } from './Container';
 import { Flex } from './Flex';
-import { Image } from './Image';
 import { LangDisplay } from './LangDisplay';
-import { Typography } from './Typography';
+import { Typography, TypographyColor } from './Typography';
 
-import { openExternalLink } from '../../assets';
-import { UtilsProps } from '../utils';
+import { OpenExternalLinkIcon } from '../../assets';
+import { BgColor, UtilsProps } from '../utils';
+import { useTheme } from '../../themes';
+
+export type ExternalLinkVariantTypes = 'disabled' | 'golden';
 
 export interface ExternalLinkProps extends UtilsProps {
+  type?: ExternalLinkVariantTypes;
+  icon?: React.ReactNode;
   text: string;
   href: string;
 }
 
 export const ExternalLink: React.FC<ExternalLinkProps> = ({
+  type,
+  icon,
   text,
   href,
   ...restProps
-}) => (
-  <Container
-    width="full"
-    $borderRadius={8}
-    px={3}
-    py="10"
-    $bgColor="input"
-    $borderWidth={1}
-    {...restProps}
-  >
-    <Flex justify="space-between" align="center" width="full">
-      <Flex align="center" gap={16}>
-        <Bullet size="sm" />
-        <Typography variant="h6" color="heading">
-          <LangDisplay text={text} />
-        </Typography>
+}) => {
+  const theme = useTheme();
+  let bgColor: BgColor = 'input';
+  let textColor: TypographyColor = 'heading';
+  if (type === 'golden') {
+    bgColor = 'golden';
+    textColor = 'black';
+  } else if (type === 'disabled') {
+    bgColor = 'slate';
+    textColor = 'disabled';
+  }
```

Comment:
Prefer maps over if else conditions.

```suggestion
  const bgColor: BgColor = bgColorMap[type] ?? 'input';
  const textColor: TypographyColor = textColorMap[type] ?? 'heading';
```